shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -252,24 +253,36 @@ private case class CloseableColumnBatchIterator(iter: Iterator[ColumnarBatch]) e
 class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
   override def supportsColumnarInput(schema: Seq[Attribute]): Boolean = true
 
-  override def supportsColumnarOutput(schema: StructType): Boolean = schema.fields.forall(f =>
+  override def supportsColumnarOutput(schema: StructType): Boolean = schema.fields.forall { f =>
     // only check spark b/c if we are on the GPU then we will be calling the gpu method regardless
-    isSupportedBySparkColumnar(f.dataType))
+    isTypeSupportedByColumnarSparkParquetWriter(f.dataType) || f.dataType == DataTypes.NullType
+  }
+
+  // Between CPU and GPU we support every type
+  private def isSchemaSupportedByParquetSerializerOnCPU(schema: StructType): Boolean = true
 
-  private def isSupportedBySparkColumnar(dataType: DataType): Boolean =
-    // AtomicType check. Everything is supported besides Structs, Maps or UDFs
+  private def isTypeSupportedByColumnarSparkParquetWriter(dataType: DataType): Boolean = {
+    // Columnar writer in Spark only supports AtomicTypes ATM
     dataType match {
       case TimestampType | StringType | BooleanType | DateType | BinaryType |
            DoubleType | FloatType | ByteType | IntegerType | LongType | ShortType => true
       case _: HiveStringType => true
```

Comment:
Have we tested with `HiveStringType`?

Reply:
As discussed offline, we should handle this as a part of a follow-on #1100